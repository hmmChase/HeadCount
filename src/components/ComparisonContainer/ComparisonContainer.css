.comparisonContainer {
  width: 100%;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: justify;
  justify-content: space-around;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}