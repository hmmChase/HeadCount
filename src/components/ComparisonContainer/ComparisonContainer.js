import React from 'react';
import './ComparisonContainer.css';
import Card from '../Card/Card';
import PropTypes from 'prop-types';


import ComparisonCard from '../ComparisonCard/ComparisonCard'
const ComparisonContainer = ({comparedDistricts, changeSelectedDistricts, comparisonObject}) => {
  const districtCards = comparedDistricts.map((district, index) => {
    return (
      <Card
        key={`Card ${index}`}
        {...district}
        changeSelectedDistricts={changeSelectedDistricts}
      />
    );
  })

  return (
    <section className="comparisonContainer">
      {(Object.keys(comparisonObject).length > 0) && <ComparisonCard comparisonObject={comparisonObject}/>}
      {districtCards}
    </section>
  )
};

ComparisonContainer.propTypes = {
  comparedDistricts: PropTypes.array.isRequired,
  changeSelectedDistricts: PropTypes.func.isRequired,
  comparisonObject: PropTypes.object.isRequired
};

export default ComparisonContainer;
