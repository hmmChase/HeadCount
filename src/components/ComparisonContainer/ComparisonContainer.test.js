import React from 'react';
import { shallow } from 'enzyme';
import ComparisonContainer from './ComparisonContainer';
const mockProps = {
  comparedDistricts: [1, 2],
  comparisonObject: {},
  changeSelectedDistricts: jest.fn()
};

describe('ComparisonContainer', () => {
  let container;
  beforeEach(() => {
    container = shallow(<ComparisonContainer {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});