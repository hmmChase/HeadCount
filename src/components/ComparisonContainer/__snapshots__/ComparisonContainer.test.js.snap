// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    changeSelectedDistricts={[Function]}
    comparedDistricts={
        Array [
            1,
            2,
          ]
    }
    comparisonObject={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Card
            changeSelectedDistricts={[Function]}
/>,
          <Card
            changeSelectedDistricts={[Function]}
/>,
        ],
        false,
      ],
      "className": "comparisonContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Card 0",
        "nodeType": "function",
        "props": Object {
          "changeSelectedDistricts": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Card 1",
        "nodeType": "function",
        "props": Object {
          "changeSelectedDistricts": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Card
              changeSelectedDistricts={[Function]}
/>,
            <Card
              changeSelectedDistricts={[Function]}
/>,
          ],
          false,
        ],
        "className": "comparisonContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Card 0",
          "nodeType": "function",
          "props": Object {
            "changeSelectedDistricts": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Card 1",
          "nodeType": "function",
          "props": Object {
            "changeSelectedDistricts": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
